### PR TITLE
Rename/move binary swap functions.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/io/byte-order.toit [hunk 1]
```diff
@@ -26,6 +26,41 @@ A byte order class with support for common read and write operations on byte
...
+  swap-16 byte-array/ByteArray -> none:
+    if byte-array.size <= 8:
+      for i := 0; i < byte-array.size; i += 2:
+        value := LITTLE-ENDIAN.uint16 byte-array i
+        BIG-ENDIAN.put-uint16 byte-array i value
+      return
+    tmp := ByteArray
+      max byte-array.size 512
+    List.chunk-up 0 byte-array.size 512: | from to size |
+      blit byte-array[from + 1..to] tmp[0..size] size/2 --source-pixel-stride=2 --destination-pixel-stride=2
```
Comment (kasperl) [inline lib/io/byte-order.toit:43]: Cache `size/2` in a local.

File: lib/io/byte-order.toit [hunk 2]
```diff
@@ -26,6 +26,41 @@ A byte order class with support for common read and write operations on byte
...
+  If the integers were in little-endian order they then are in big-endian byte order
+  If the integers were in big-endian order they then are in little-endian byte order.
+  */
+  swap-32 byte-array/ByteArray -> none:
+    tmp := ByteArray
+      max byte-array.size 512
+    List.chunk-up 0 byte-array.size 512: | from to size |
+      slice := byte-array[from..to]
+      buffer := tmp[..size]
+      blit slice[3..] buffer      size/4 --source-pixel-stride=4 --destination-pixel-stride=4
```
Comment (kasperl) [inline lib/io/byte-order.toit:58]: Cache `size/4` in a local.

File: lib/io/byte-order.toit [hunk 3]
```diff
@@ -26,6 +26,41 @@ A byte order class with support for common read and write operations on byte
...
+      byte-array.replace from tmp 0 size
+
+  /**
+  Swaps the byte-order of all 32-bit integers in the $byte-array.
+  If the integers were in little-endian order they then are in big-endian byte order
+  If the integers were in big-endian order they then are in little-endian byte order.
+  */
+  swap-32 byte-array/ByteArray -> none:
+    tmp := ByteArray
+      max byte-array.size 512
```
Comment (kasperl) [inline lib/io/byte-order.toit:54]:
```suggestion
    tmp := ByteArray
        max byte-array.size 512
```

File: lib/io/byte-order.toit [hunk 4]
```diff
@@ -26,6 +26,41 @@ A byte order class with support for common read and write operations on byte
...
+  If the integers were in big-endian order they then are in little-endian byte order.
+  */
+  swap-16 byte-array/ByteArray -> none:
+    if byte-array.size <= 8:
+      for i := 0; i < byte-array.size; i += 2:
+        value := LITTLE-ENDIAN.uint16 byte-array i
+        BIG-ENDIAN.put-uint16 byte-array i value
+      return
+    tmp := ByteArray
+      max byte-array.size 512
```
Comment (kasperl) [inline lib/io/byte-order.toit:41]:
```suggestion
    tmp := ByteArray
        max byte-array.size 512
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/io/byte-order.toit (hunk 4 above)
Comment (floitsch) [reply lib/io/byte-order.toit:41]: Will do this in a follow-up PR. Here I just move the functions.

File: lib/io/byte-order.toit (hunk 1 above)
Comment (floitsch) [reply lib/io/byte-order.toit:43]: Will do this in a follow-up PR. Here I just move the functions.

File: lib/io/byte-order.toit (hunk 3 above)
Comment (floitsch) [reply lib/io/byte-order.toit:54]: Will do this in a follow-up PR. Here I just move the functions.

File: lib/io/byte-order.toit (hunk 2 above)
Comment (floitsch) [reply lib/io/byte-order.toit:58]: Will do this in a follow-up PR. Here I just move the functions.

**Comment by floitsch** [comment]
Feedback is addressed in https://github.com/toitlang/toit/pull/2169

**Pull request merged**